dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+            query_stop_time = datetime.utcnow() - self.lag
+            query_start_time = query_stop_time - self.period
+
+            self.logger.info(
+                "Searching for transits from %s to %s"
+                % (str(query_start_time), str(query_stop_time))
+            )
+
+            window = self.nsigma * cal_utils.guess_fwhm(
+                400.0, pol="X", dec=src_body.dec.radians, sigma=True
+            )
+            # nsigma distance in degree from transit from peak
+            time_delta = (
+                2 * (window / 360.0) * 24 * 3600.0
+            )  # time window around transit in sec
+
+            # Find all calibration files that have transit of given source
+            f = self.Finder()
+            f.set_time_range(query_start_time, query_stop_time)
+            f.accept_all_global_flags()
+            f.only_corr()
+            f.filter_acqs((data_index.ArchiveInst.name == self.acq_suffix))
+            f.include_transits(src_body, time_delta=time_delta)
+            file_list = f.get_results()
+            nfile = len(file_list)
+            times = [file_list[ii][1] for ii in range(0, nfile)]
+
+            try:
+                all_files = file_list[0][0]
+                if not all_files:
+                    raise IndexError()
+            except IndexError:
+                tmp += "No {} files found from last {} for source transit {}.\n".format(
+                    self.acq_suffix, self.period, src
+                )
+                err_msg += tmp
+                self.logger.info(tmp)
+                continue
+
+            # Loop over files
+            for file_index, files in enumerate(all_files):
+                # Read file time range
+                with h5py.File(files, "r") as handler:
+                    file_timestamp = handler["index_map"]["time"]["ctime"][:]
+
+                csd = int(np.median(ephemeris.csd(file_timestamp)))
+
+                # Compute source coordinates
+                src_ra, src_dec = ephemeris.object_coords(
+                    src_body, date=np.median(file_timestamp), deg=False
+                )
+                source_start_time, source_stop_time = file_list[file_index][1]
+
+                is_daytime = 0.0
+                # test if the source is transiting in the daytime
+                solar_rise = ephemeris.solar_rising(
+                    source_start_time - 24.0 * 3600.0, end_time=source_stop_time
+                )
+                for rr in solar_rise:
+                    ss = ephemeris.solar_setting(rr)[0]
+                    if (source_start_time <= ss) and (rr <= source_stop_time):
+                        is_daytime += 1
+                        tt = ephemeris.solar_transit(rr)[0]
+                        # if Sun is in the beam
+                        if source_start_time <= tt <= source_stop_time:
+                            is_daytime += 1
+                        break
+
+                if is_daytime > 1:
```

Comment:
The 1 here should probably be a config parameter with the following interpretation:

- 0 would only process sources that transit at night
- 1 would process all sources that transit when the sun is outside of the primary beam
- 2 would process all sources

I would then save the `is_daytime` variable as an attribute of the output file, so that the user can easily filter on this when examining the results.